File: Excel/800-899/833/833 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/800-899/833/833 Summarize.xlsx"
input = read_excel(path, range = "A2:C14")
test  = read_excel(path, range = "E2:H5")

result = input %>%
  mutate(Date = ifelse(!is.na(Task), Data, NA)) %>%
  fill(Date, Task) %>%
  filter(Data != Date) %>%
  summarise(Amount = sum(Data, na.rm = TRUE), .by = c(Store, Date, Task)) %>%
  mutate(Date = janitor::excel_numeric_to_date(Date) %>% as.POSIXct())

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE

File: Excel/800-899/833/833 Challenge.py
import pandas as pd

path = "800-899/833/833 Summarize.xlsx"
input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=13)
test = pd.read_excel(path, usecols="E:H", skiprows=1, nrows=3).rename(columns=lambda col: col.replace('.1', ''))

input['Date'] = input.apply(lambda row: row['Data'] if pd.notna(row['Task']) else pd.NA, axis=1)
input['Date'] = input['Date'].fillna(method='ffill')
input['Task'] = input['Task'].fillna(method='ffill')
filtered = input[input['Data'] != input['Date']]

result = (
    filtered
    .groupby(['Store', 'Date', 'Task'], as_index=False)
    .agg(Amount=('Data', lambda x: x.sum(skipna=True)))
)
print(result.equals(test)) # True